components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.ts

```diff
@@ -0,0 +1,335 @@
+    .pipe(
+      takeUntil(this.isDestroyed),
+      filter(state => state === EntityStatus.loadingSuccess))
+      .subscribe(state => {
+        this.creating = false;
+        if (state === EntityStatus.loadingSuccess) {
+          this.store.dispatch(new GetEnvironments(payload));
+          this.closeCreateModal();
+        }
+      });
+
+    combineLatest([
+      this.store.select(saveStatus),
+      this.store.select(saveError)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      filter(([state, error]) => state === EntityStatus.loadingFailure && !isNil(error)))
+      .subscribe(([_, error]) => {
+        if (error.status === HttpStatus.CONFLICT) {
+          this.conflictErrorEvent.emit(true);
+          this.conflictError = true;
+        } else {
+          this.store.dispatch(new GetEnvironments(payload));
+          this.creating = false;
+
+          // Close the modal on any other error because it will be displayed in the banner.
+          this.closeCreateModal();
+        }
+        this.creating = false;
+      });
+  }
+
+  public handleInput(event: KeyboardEvent): void {
+    if (this.isNavigationKey(event)) {
+      return;
+    }
+    this.conflictError = false;
+  }
+
+  tabChange(tab: number) {
+    // Tab indices here correspond with the order of
+    // `<app-infra-tab-change>` elements in the template.
+    switch (tab) {
+      case 0:
+        this.telemetryService.track(CREATE_TAB_NAME, 'details');
+        this.resetTabs();
+        this.detailsTab = true;
+        break;
+      case 1:
+        this.telemetryService.track(CREATE_TAB_NAME, 'constraints');
+        this.resetTabs();
+        this.constraintsTab = true;
+        break;
+      case 2:
+        this.telemetryService.track(CREATE_TAB_NAME, 'default');
+        this.resetTabs();
+        this.defaultTab = true;
+        break;
+      case 3:
+        this.telemetryService.track(CREATE_TAB_NAME, 'override');
+        this.resetTabs();
+        this.overrideTab = true;
+        break;
+    }
+  }
+
+
+  // Handles the data of cookbook version array coming from constraint tab.
+  constraintItemsHandler(value: Array<DynamicGrid> = []    ) {
+    this.constraintArray = value;
+  }
+
+  // Getting list of cookbook names
+  loadCookbookConstraint() {
+    this.name_id = '';
+    this.store.dispatch(new GetCookbooks({
+      server_id: this.serverId, org_id: this.orgId
+    }));
+
+    combineLatest([
+      this.store.select(getAllCookbooksForOrgStatus),
+      this.store.select(allCookbooks)
+    ]).pipe(takeUntil(this.isDestroyed))
+    .subscribe(([ getCookbooksSt, allCookbooksState]) => {
+      if (getCookbooksSt === EntityStatus.loadingSuccess && !isNil(allCookbooksState)) {
+        this.constraintKeys = [];
+        this.cookbooks = allCookbooksState;
+        this.cookbooks.forEach((cookbook) => {
+          this.constraintKeys.push(cookbook.name);
+        });
+      }
+      this.name_id = this.constraintKeys[0];
```

Comment:
Is it valid to always use the first one? Would be helpful to have an in-code comment explaining that.

Reply:
yes, when we click on create environment pop-up then, by default the dropdown value of the constraint name set to be 1st value of constraint keys.